lib/primer/classify/cache.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+module Primer
+  class Classify
+    # :nodoc:
+    class Cache
+      # rubocop:disable Style/MutableConstant
+      LOOKUP = {}
+      # rubocop:enable Style/MutableConstant
+
+      class <<self
+        def read(memo, key, val, breakpoint)
+          value = LOOKUP.dig(breakpoint, key, val)
+          memo[:classes] << value if value
+        end
+
+        def clear!
+          LOOKUP.clear
+        end
+
+        def preload!
+          preload(
+            keys: Primer::Classify::MARGIN_DIRECTION_KEYS,
+            values: [-6, -5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5, 6]
+          )
+
+          preload(
+            keys: (Primer::Classify::SPACING_KEYS - Primer::Classify::MARGIN_DIRECTION_KEYS),
+            values: [0, 1, 2, 3, 4, 5, 6]
+          )
+
+          preload(
+            keys: Primer::Classify::DIRECTION_KEY,
+            values: [:row, :column]
+          )
+
+          preload(
+            keys: Primer::Classify::JUSTIFY_CONTENT_KEY,
+            values: [:flex_start, :flex_end, :center, :space_between, :space_around]
+          )
+
+          preload(
+            keys: Primer::Classify::ALIGN_ITEMS_KEY,
+            values: [:flex_start, :flex_end, :center, :baseline, :stretch]
+          )
+
+          preload(
+            keys: Primer::Classify::DISPLAY_KEY,
+            values: [:flex, :block, :inline_block, :inline_flex, :none, :table, :table_cell]
+          )
+
+          preload(
+            keys: [Primer::Classify::COLOR_KEY, Primer::Classify::BG_KEY],
+            values: [:blue, :gray_dark, :gray, :gray_light, :red, :orange, :orange_light, :yellow, :green, :purple, :white, :pink]
```

Comment:
How did you arrive on these `values` here? (And for the rest of this file?) I'd love to have a process to ensure this optimization stays up to date 😄

Reply:
It was very unscientific, we went through https://primer.style/css/ and tried to find colors to fill in here 😅.

There's probably a few better ways to handle it, like maybe parsing primer itself for color variables that we could use but we didn't have time to look into anything like that.